src/coreclr/jit/gentree.cpp

```diff
@@ -9808,6 +9808,33 @@ void Compiler::gtDispNode(GenTree* tree, IndentStack* indentStack, _In_ _In_opt_
                         layout = varDsc->GetLayout();
                     }
                 }
+                else if (tree->OperIs(GT_INDEX))
+                {
+                    GenTreeIndex* asInd = tree->AsIndex();
+                    if (asInd->gtStructElemClass != nullptr)
+                    {
+                        // We could create a layout with `typGetObjLayout(asInd->gtStructElemClass)` but that
+                        // would potentially impact codegen by adding something to the layout table.
```

Comment:
It seems unthinkable to me that adding a layout to the table could influence the actual machine code produced.

(I do agree this is the right thing to do though, as it makes SPMI less likely to fail with a "missing" error due to the lack of GC layout maps)

Reply:
> It seems unthinkable to me that adding a layout to the table could influence the actual machine code produced.

true, but it's an important principle that debug/dump routines should never affect data structures used by non-debug/dump compilation.